### PR TITLE
Add components and parames to whatsapp template translations

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/utils/whatsapp/tasks.py [hunk 1]
```diff
@@ -131,6 +163,9 @@ def update_local_templates(channel, templates_data):
             status = TemplateTranslation.STATUS_UNSUPPORTED_LANGUAGE
             language = template["language"]
 
+        if not content_parts or not all_supported:
+            status = TemplateTranslation.STATUS_UNSUPPORTED_COMPONENTS
```
Comment (norkans7) [inline temba/utils/whatsapp/tasks.py:167]: this status is a way for now so we can finish other changes to allow supporting all the templates components

File: temba/utils/whatsapp/tasks.py [hunk 2]
```diff
@@ -89,6 +90,36 @@ def _calculate_variable_count(content):
...
+            else:
+                params[component_type].append({"type": component["format"].lower()})
+        if component_type == "body":
+            for match in VARIABLE_RE.findall(component["text"]):
+                params[component_type].append({"type": "text"})
+        if component_type == "buttons":
+            buttons = component["buttons"]
+            for button in buttons:
+                if button["type"].lower() in ["quick_reply", "phone_number"]:
+                    params[component_type].append({"type": ""})
```
Comment (norkans7) [inline temba/utils/whatsapp/tasks.py:112]: buttons is a list of button and to not loose the order/index of the button requiring the variable parameters, I add the `"type": ""` for those that do not have any variable

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:112]: That way we keep the exact number for buttons parameters in the list

Comment (rowanseymour) [reply temba/utils/whatsapp/tasks.py:112]: hmm `type=""` doesn't seem very intuitive... maybe dict keys like `button.1` ?

File: temba/utils/whatsapp/tasks.py (hunk 1 above)
Comment (rowanseymour) [reply temba/utils/whatsapp/tasks.py:167]: or can just we just strip out BUTTON components for now?

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:167]: we now have them in the components field only that we are not yet rely on
And the content field which we rely on ignores the buttons

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:167]: previously we were not creating any template at all https://github.com/nyaruka/rapidpro/pull/4977/files#diff-7741e473d78645add652703a29785cea63138070bf17c75934ca055616da5456L121-L123

Now I think we want to save that but not usable in the flows yet

File: temba/utils/whatsapp/tests.py [hunk 3]
```diff
@@ -152,12 +196,88 @@ def test_update_local_templates_whatsapp(self):
...
+                    ],
+                },
+            ],
+            ct.components,
+        )
+        self.assertEqual(
+            {
+                "header": [{"type": "image"}],
+                "body": [{"type": "text"}, {"type": "text"}],
+                "buttons": [{"button.3": [{"type": "text"}]}],
```
Comment (rowanseymour) [inline temba/utils/whatsapp/tests.py:277]: I meant something like... 

```json
{
    "header": [{"type": "image"}],
    "body": [{"type": "text"}, {"type": "text"}],
    "button.3": [{"type": "text"}],
}
```

So it's always just location -> list of params

Comment (norkans7) [reply temba/utils/whatsapp/tests.py:277]: I tried to match the exact components JSON structure which has the list of buttons nested inside the `buttons` component

Comment (norkans7) [reply temba/utils/whatsapp/tests.py:277]: Okay

Comment (norkans7) [reply temba/utils/whatsapp/tests.py:277]: I made the adjustment in https://github.com/nyaruka/rapidpro/pull/4979/files and merged that in this branch

**Pull request merged**